src/Nevermind/Nevermind.Core/Crypto/Random.cs

```diff
@@ -6,7 +6,14 @@ public static class Random
 
         public static byte[] GeneratePrivateKey()
         {
-            byte[] bytes = new byte[32];
+            var bytes = new byte[32];
+            SecureRandom.GetBytes(bytes);
+            return bytes;
+        }
+
+        public static byte[] GenerateRandomBytes(int lenght)
+        {
```

Comment:
if not behind interface then equally we can use SecureRandom.GetBytes directly, otherwise let us push it behind ISecureRandom so we can test with this class wherever used